### ISSUE TITLE
makeTerraformInputs converts empty sets wrong

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi-terraform-bridge#1970]
### What happened?

Similar to https://github.com/pulumi/pulumi-terraform-bridge/issues/1916 and https://github.com/pulumi/pulumi-terraform-bridge/issues/1915, empty sets are also represented wrong in pulumi.

We output a `cty.NullVal` instead of a `cty.SetValEmpty`

### Example

`TestOptionalSetNotSpecified`

### Output of `pulumi about`

.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by t0yv0** [comment]
Curious if we can fix this. There are some good changes blocked on this now, including https://github.com/pulumi/pulumi-terraform-bridge/pull/1976

**Comment by VenelinMartinov** [comment]
Yup, actively working on this in https://github.com/pulumi/pulumi-terraform-bridge/pull/1971
